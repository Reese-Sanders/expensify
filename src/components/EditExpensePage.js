import React from 'react';
import { useLocation, useNavigate, useParams } from 'react-router-dom';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
  let location = useLocation();
  let { id } = useParams();
  let expense = props.expenses.find((expense) => expense.id === id);
 
  let navigate = useNavigate(); 
  const routeChange = () =>{ 
    let path = `/`; 
    navigate(path);
  }


  return (
    <div>
      <ExpenseForm 
        expense={expense}
        onSubmit={(expenseEdit) => {
          props.dispatch(editExpense(expense.id, expenseEdit));
          routeChange();
        }}
      />
      <button
      onClick={() => {
        props.dispatch(removeExpense({id: expense.id}));
        routeChange();
      }}
      >
        Remove
      </button>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    expenses: state.expenses
  }
}

export default connect(mapStateToProps)(EditExpensePage);