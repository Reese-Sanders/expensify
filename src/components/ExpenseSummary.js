import React from "react";
import { connect } from "react-redux";
import selectExpenses from '../selectors/expenses';
import expensesTotal from "../selectors/expenses-total";
import numeral from "numeral";


export const ExpenseSummary = (props) => (
  <div>
    <p>
    Viewing {props.expenses.length?props.expenses.length:0} totalling {numeral(expensesTotal(props.expenses)/100).format('$0,0.00')} 
    </p>
  </div>
);

// const mapStateToProps = (state) => ({ 
//   expenses: state.expenses,
//   filters: state.filters
// });

const mapStateToProps = (state) => ({ 
  expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseSummary);