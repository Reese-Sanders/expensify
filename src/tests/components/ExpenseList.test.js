import React from 'react';
import { render, screen } from '@testing-library/react';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';
import { BrowserRouter } from 'react-router-dom';

const MockExpenseList = (props) => {
  return (
    <BrowserRouter>
      <ExpenseList expenses={props.expenses}/>
    </BrowserRouter>
  )
}

test('should render ExpenseList with expenses', () => {
  const container = render(<MockExpenseList expenses={expenses}/>);
  expect(container).toMatchSnapshot();
});

test('should render ExpenseList with no expenses', () => {
  const container = render(<MockExpenseList expenses={[]}/>);
  expect(container).toMatchSnapshot();
});