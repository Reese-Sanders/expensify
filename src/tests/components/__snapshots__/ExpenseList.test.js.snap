// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <a
            href="/edit/1"
          >
            <h3>
              Gum
            </h3>
          </a>
          <p>
            195
             - 
            0
          </p>
        </div>
        <div>
          <a
            href="/edit/2"
          >
            <h3>
              Rent
            </h3>
          </a>
          <p>
            109500
             - 
            -345600000
          </p>
        </div>
        <div>
          <a
            href="/edit/3"
          >
            <h3>
              Credit Card
            </h3>
          </a>
          <p>
            4500
             - 
            345600000
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <a
          href="/edit/1"
        >
          <h3>
            Gum
          </h3>
        </a>
        <p>
          195
           - 
          0
        </p>
      </div>
      <div>
        <a
          href="/edit/2"
        >
          <h3>
            Rent
          </h3>
        </a>
        <p>
          109500
           - 
          -345600000
        </p>
      </div>
      <div>
        <a
          href="/edit/3"
        >
          <h3>
            Credit Card
          </h3>
        </a>
        <p>
          4500
           - 
          345600000
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render ExpenseList with no expenses 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p>
          No Expenses
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        No Expenses
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
